win32/Lib/win32timezone.py

```diff
@@ -596,11 +596,27 @@ def __str__(self):
         return self.displayName
 
     def tzname(self, dt):
-        winInfo = self.getWinInfo(dt)
-        if self.dst(dt) == winInfo.daylight_bias:
+        """
+        >>> MST = TimeZoneInfo('Mountain Standard Time')
+        >>> MST.tzname(datetime.datetime(2003, 8, 2))
+        'Mountain Daylight Time'
+        >>> MST.tzname(datetime.datetime(2003, 11, 25))
+        'Mountain Standard Time'
+        """
+
+        winInfo = self.getWinInfo(dt.year)
+        if self.dst(dt) == -winInfo.daylight_bias:
```

Comment:
I just noticed, but probably better not to repeat `self.dist` call up to 3 times.

Reply:
I'm a little reluctant to spend energy optimizing a function that's never been used. It's also unnecessary to address this bugfix. Let's leave it for another contrib (be my guest).